ClosedXML/Excel/CalcEngine/Expression.cs

```diff
@@ -125,6 +125,12 @@ public virtual Expression Optimize()
                 return ts.TotalDays;
             }
 
+            // handle string
+            if (v is string s && double.TryParse(s, out var doubleValue))
```

Comment:
Should it be culture specific? Excel, I expect, uses current locale, do we have to do the same? @igitur, your opinion? 
Either way, better to specify locale explicitly and cover such cases with tests. For example, 1,000 is one thousand in en-US and one in ru-RU.

Reply:
TryParse uses the current locale by default.

Reply:
Yeah, I know, but to avoid confusion during possible future refactoring it is nice to have the information about the intention: if default locale was used accidentally or on purpose. In many other places we do not rely on current locale so rare exceptions must be emphasized. IMO

Reply:
This relates to a comment I made somewhere in another PR. I think we should optionally pass through the `CultureInfo` instance in the `XLWorkbook` instance and use that everywhere where applicable. Default to `CurrentCulture`. We'll have to do a separate PR for that.